tools/sesman/sesmanworker/engine.py

```diff
@@ -1760,3 +1774,198 @@ def get_target_str(self):
+    duplicated filter keywords, missing values and others aren't 
+    allowed and will be skipped and considered as no valid.
+    The following examples are forbidden :
+    "$account:my_account,$account:my_account"
+    "$domain:my_domain,"
+    "$my_device:my_device"
+    "$service:"
+    "$abcd:my_author"
+
+    """
+    
+    __slots__ = ("filter_pattern_dict")
+    
+    def __init__(self) :
+        self.filter_pattern_dict = {
+            "account" : None,
+            "domain" : None,
+            "device" : None,
+            "service" : None,
+            "auth" : None
+        }
+
+    def parse_target_filter_pattern(self, target_filter_pattern) :
+        FILTER_KV_SEPARATOR = ':'
+        FILTER_SEPARATOR = ','
+        filter_kv_list = target_filter_pattern.split(FILTER_SEPARATOR)
+
+        for filter_kv in filter_kv_list :
+            token_list = filter_kv.split(FILTER_KV_SEPARATOR, 1)
+
+            if len(token_list) != 2 :
+                raise ParsingError(filter_kv)
+
+            filter_keyword, filter_value = tuple(token_list)
+
+            if not filter_value :
+                raise ParsingError(filter_kv)
+
+            is_filter_keyword, filter_keyword_no_prefix = (
+                self._is_filter_keyword(filter_keyword))
+
+            if not is_filter_keyword :
+                raise FilterKeywordSyntaxError(filter_keyword)
+            if self.filter_pattern_dict[filter_keyword_no_prefix] is None :
+                self.filter_pattern_dict.update(
+                    {filter_keyword_no_prefix : filter_value})
+            else :
+                raise FilterValueDuplicateError(filter_keyword)
+
+    def is_filterable(self, target) :
+        if not self.filter_pattern_dict.values() :
+            raise NoFilterValueError(target)
+
+        target_field_dict = {}
+
+        target_field_dict.fromkeys(range(len(self.filter_pattern_dict)))
+        self._get_fields_from_target(target, target_field_dict)
+        for filter_keyword, filter_value in self.filter_pattern_dict.items() :
+            if filter_value is None :
+                continue
+            if not self._filter_pattern_is_applicable(filter_keyword,
+                                                      filter_value,
+                                                      target_field_dict) :
+                return False
+        return True
+
+    def _is_filter_keyword(self, filter_keyword) :
+        if len(filter_keyword) < 2 or filter_keyword[0] != '$' :
+            return False, None
+
+        filter_keyword_no_prefix = filter_keyword[1:]
+
+        return ((True, filter_keyword_no_prefix)
+                if filter_keyword_no_prefix in self.filter_pattern_dict
+                else (False, None))
+
+    def _get_fields_from_target(self, target, target_field_dict) :
+        SEPARATORS = '@'
+        escaped_seps = re.escape(SEPARATORS)
+        sep_count = len(re.findall("[%s]" % escaped_seps, target))
```

Comment:
```py
sep_count = len(re.findall("[@]", target))
# =>
sep_count = target.count('@')
```
:thinking:

Reply:
The fact is I used "len(re.findall())" and not "target.count()" is because if later, we want to add another separator in SEPARATORS variable like '+' (and thus "@+"), we won't need to change _get_field_from_target() implementation. 

With target.count() which count the number of occurence of a string and NOT the number of occurrence of each char of the string, we will force us to refactorize if we add new separator